src/viam/sdk/services/mlmodel/server.cpp

```diff
@@ -64,92 +60,39 @@ ::grpc::Status MLModelServiceServer::Infer(
         extra = struct_to_map(request->extra());
     }
 
-    if (request->has_input_data()) {
-        // TODO: We need shape and type information from the metadata to
-        // be able to unpack input tensors. It would be nice to find some
-        // way to cache this information. This isn't exactly the same
-        // issue as the client side (see RSDK-3298), but is closely
-        // related.
-        const auto md = mlms->metadata(extra);
-
-        mlmodel_details::tensor_storage input_storage;
-        MLModelService::named_tensor_views inputs;
-
-        const auto& input_fields = request->input_data().fields();
-        for (const auto& input : md.inputs) {
-            const auto where = input_fields.find(input.name);
-
+    const auto md = mlms->metadata(extra);
```

Comment:
Git didn't do a great job with this diff. Is it safe to assume that this is simply a removal of the `has_input_data` path with no meaningful changes to the `has_input_tensors` path?

Reply:
Yep!